src/zope/testrunner/tb_format.py

```diff
@@ -20,9 +20,34 @@
 import zope.testrunner.feature
 
 
+try:
+    _iter_chain = traceback._iter_chain
+except AttributeError:
+    # Python 3.5
+    def _iter_chain(exc, custom_tb=None, seen=None):
+        if seen is None:
+            seen = set()
+        seen.add(exc)
+        its = []
+        cause = exc.__cause__
+        if cause is not None and cause not in seen:
+            its.append(_iter_chain(cause, None, seen))
+            its.append([(traceback._cause_message, None)])
+        else:
+            context = exc.__context__
+            if context is not None and context not in seen:
+                its.append(_iter_chain(context, None, seen))
+                its.append([(traceback._context_message, None)])
+        its.append([(exc, custom_tb or exc.__traceback__)])
+        # itertools.chain is in an extension module and may be unavailable
+        for it in its:
+            for x in it:
+                yield x
```

Comment:
Looks like I looked up a somewhat outdated copy of that file, sorry for this. Link to the copy I used: http://svn.python.org/projects/python/branches/pep-0384/Lib/traceback.py

I'm going to update the PR using a more recent version (with a little adjustment to the "yield from" part, as that's an invalid syntax for Python 2)